src/frame/decompress.rs

```diff
@@ -19,6 +19,30 @@ use crate::block::WINDOW_SIZE;
 /// This reader can potentially make many small reads from the underlying
 /// stream depending on its format, therefore, passing in a buffered reader
 /// may be beneficial.
```

Comment:
We can internally always wrap the `Read` with a `BufReader`. Forgetting to wrap something in a `BufReader` is a common performance issue I think

Reply:
TBH I copied this comment from the snap crate so it might very well be wrong as there's only 2 reads per block. 1 for the block header and another one to get its full contents.

Either way it's only really applicable to a file so we shouldn't force the overhead onto correct usages.

Reply:
I got these
```
FrameDecompress/lz4_flex_rust_buffered/725                                                                             
                        time:   [2.9677 us 2.9788 us 2.9905 us]
                        thrpt:  [231.20 MiB/s 232.11 MiB/s 232.98 MiB/s]
Found 16 outliers among 100 measurements (16.00%)
  4 (4.00%) low mild
  1 (1.00%) high mild
  11 (11.00%) high severe
FrameDecompress/lz4_flex_rust_raw/725                                                                             
                        time:   [3.8128 us 3.8289 us 3.8466 us]
                        thrpt:  [179.75 MiB/s 180.58 MiB/s 181.34 MiB/s]
Found 13 outliers among 100 measurements (13.00%)
  3 (3.00%) low mild
  7 (7.00%) high mild
  3 (3.00%) high severe
FrameDecompress/lz4_flex_rust_buffered/34308                                                                             
                        time:   [20.215 us 20.423 us 20.646 us]
                        thrpt:  [1.5476 GiB/s 1.5645 GiB/s 1.5806 GiB/s]
Found 10 outliers among 100 measurements (10.00%)
  6 (6.00%) high mild
  4 (4.00%) high severe
FrameDecompress/lz4_flex_rust_raw/34308                                                                             
                        time:   [20.474 us 20.750 us 21.049 us]
                        thrpt:  [1.5180 GiB/s 1.5398 GiB/s 1.5606 GiB/s]
Found 6 outliers among 100 measurements (6.00%)
  4 (4.00%) high mild
  2 (2.00%) high severe
FrameDecompress/lz4_flex_rust_buffered/66675                                                                             
                        time:   [21.209 us 21.547 us 21.939 us]
                        thrpt:  [2.8304 GiB/s 2.8819 GiB/s 2.9278 GiB/s]
Found 4 outliers among 100 measurements (4.00%)
  3 (3.00%) high mild
  1 (1.00%) high severe
FrameDecompress/lz4_flex_rust_raw/66675                                                                             
                        time:   [21.430 us 21.636 us 21.872 us]
                        thrpt:  [2.8390 GiB/s 2.8700 GiB/s 2.8977 GiB/s]
Found 4 outliers among 100 measurements (4.00%)
  4 (4.00%) high mild
```

So it only makes a difference for the tiny input. But the difference is still less than a microsecond, which is nothing when dealing with real files. So maybe that comment should be removed.

Reply:
Pushed a6fac63